packages/next-server/lib/dynamic.js

```diff
@@ -84,11 +84,9 @@ export default function dynamic (dynamicOptions, options) {
 
   // support for disabling server side rendering, eg: dynamic(import('../hello-world'), {ssr: false})
   if (typeof loadableOptions.ssr === 'boolean') {
-    if (!loadableOptions.ssr) {
-      delete loadableOptions.ssr
+    if (!(delete loadableOptions.ssr)) {
```

Comment:
Thanks for the feedback on the previous PR. I was really introducing an error cause ssr true would cause a problem in the if.

I suggest taking advantage of delete's return.
delete will execute the operation and return true or false, so we can do it and then call noSSR() or not.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete#Return_value

Let me know if you agree @timneutkens